query-builder/src/main/java/com/datastax/oss/driver/api/querybuilder/schema/CreateKeyspace.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.api.querybuilder.schema;
+
+import com.datastax.oss.driver.api.querybuilder.BuildableQuery;
+import com.google.common.collect.ImmutableMap;
+import java.util.Map;
+
+public interface CreateKeyspace extends BuildableQuery {
```

Comment:
For DML statements I tried to avoid returning `BuildableQuery` until the query is complete, so technically we would need to wait for a replication strategy if we wanted to do this here. But this is kind of complicated because the properties can also be provided with the generic method.
Maybe this is overkill for DDL statements.

Reply:
in this particular case it seems like it could be reasonable to do it that way.  As I do more DML implementations I'll see how reasonable this would be.  For a lot of it i suspect that there's only one thing absolutely required.  I.E. for create table you need at least 1 partition key.